src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs

```diff
@@ -0,0 +1,268 @@
+﻿using AutoFixture.MSTest;
+using EPR.Payment.Facade.Common.Configuration;
+using EPR.Payment.Facade.Common.Constants;
+using EPR.Payment.Facade.Common.Dtos.Request.RegistrationFees;
+using EPR.Payment.Facade.Common.Dtos.Response.RegistrationFees;
+using EPR.Payment.Facade.Common.Exceptions;
+using EPR.Payment.Facade.Common.RESTServices.RegistrationFees;
+using EPR.Payment.Facade.Common.UnitTests.TestHelpers;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Moq;
+using Moq.Protected;
+using Newtonsoft.Json;
+using System.Net;
+using System.Text;
+
+namespace EPR.Payment.Facade.Common.UnitTests.RESTServices
+{
+    [TestClass]
+    public class HttpRegistrationFeesServiceTests
+    {
+        private Mock<IHttpContextAccessor> _httpContextAccessorMock = null!;
+        private Mock<IOptions<Service>> _configMock = null!;
+        private ProducerRegistrationFeesRequestDto _producerRegistrationFeesRequestDto = null!;
+        private RegistrationFeesResponseDto _registrationFeesResponseDto = null!;
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            // Mock configuration
+            var config = new Service
+            {
+                Url = "https://api.example.com",
+                EndPointName = "registration-fees",
+                HttpClientName = "HttpClientName"
+            };
+
+            _configMock = new Mock<IOptions<Service>>();
+            _configMock.Setup(x => x.Value).Returns(config);
+
+            _httpContextAccessorMock = new Mock<IHttpContextAccessor>();
+            _producerRegistrationFeesRequestDto = new ProducerRegistrationFeesRequestDto
+            {
+                ProducerType = "L",
```

Comment:
isn't the producer type "large"? we are using 3 values - small, large or empty string